### PR TITLE
testing git-diff

### PR DESCRIPTION
**Pull request by relentless-pursuit**
Part of #1

### CODE REVIEW

**Comment by relentless-pursuit** [comment]
I have been reading about this CI command added git-diff, but struggle to understand. To my understanding, CI checks out my PR and compares the changes with the latest code available on remote. And, if any untracked changes, then it will fail. For instance, i have untracked changes in this PR in my local. But, the job seems to succeed.

**Comment by romani** [comment]
https://github.com/checkstyle/test-configs/actions/runs/9578108855/job/26407651096?pr=40#step:5:204

test is not failed, even it is crashed with exception and produced no output.

I checked it on local, no diff on local.
diff check works.

**Comment by romani** [comment]
you will improve at scope of this issue - https://github.com/checkstyle/test-configs/issues/41

**Comment by romani** [comment]
@relentless-pursuit , please rebase

**Comment by relentless-pursuit** [comment]
Rebased. But, i am not sure how this git diff works? Shouldn't the job be pointing to new files added liek clone the latest checkstyle test-config and compare with the PR?

**Comment by romani** [comment]
No, it just do git status on PR code after execution

**Comment by relentless-pursuit** [comment]
> No, it just do git status on PR code after execution.
> 
> Create new file in git repository from test execution and it should be caught

yes, i did. but, i guess i didn't see any failure. i created a folder from mainslaunchertest.java and then i first kept them in unstaged. but, the job didn't fail. then in the next commit, i put them in the staging and did commit.

**Comment by romani** [comment]
New empty folder is not status changed for git, please create file

File: AbstractClassName/Example1/config.xml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0"?>
+<!DOCTYPE module PUBLIC
```
Comment (romani) [inline AbstractClassName/Example1/config.xml:2]: Remove from commit this folder and it should be caught

**Review by romani — CHANGES_REQUESTED** [review]
Item

**Comment by relentless-pursuit** [comment]
> New empty folder is not status changed for git, please create file

I had actually created a complete folder with config.xml. Except, i didn't commit but kept in unstaged env. Yes, i will act on the review and test again.

**Comment by romani** [comment]
It works, see it here https://github.com/checkstyle/test-configs/actions/runs/9635984919/job/26573394995?pr=53#step:6:30

I think you can close this PR

**Comment by relentless-pursuit** [comment]
Closing the PR as testing is validated.

**Pull request closed without merging**